src/Formula/Parsing.hs

```diff
@@ -296,13 +298,15 @@ instance Parse PickInst where
     where
       instParse = do
         string "PickInst("
-        cs <- parser
+        tokenSymbol "["
+        trees' <- lexeme (many (noneOf ",]")) `sepBy` char ','
+        tokenSymbol "]"
```

Comment:
After such ad hoc changes to parsers etc., I would feel safer if there was at least some unit test of the kind "parsing and pretty-printing properly round-trip on `dPickInst`".